Makefile

```diff
@@ -105,7 +105,9 @@ CFLAGS  = -msgstyle gcc \
 		-maxerrors $(MAX_ERRORS)
 
 ifeq ($(NON_MATCHING),1)
-CFLAGS += -DNON_MATCHING
+	CFLAGS += -DNON_MATCHING
+else
+	CFLAGS += -DMUST_MATCH
```

Comment:
Both imply `__MWERKS__ && GEKKO`, but differ in whether the target must match. Honestly I'd like to phase out `NON_MATCHING` because by definition it implies a target architecture and compiler, but semantically any build on other platforms would be "non-matching." Now that we care about target platform, `!NON_MATCHING` no longer equals `MUST_MATCH`.

Reply:
But `!NON_MATCHING` is literally `MUST_MATCH`, that's just how it works. If you're targeting any other platform then your build is not matching.

Reply:
The point I was trying to make is only the MWCC script is defining `NON_MATCHING`. When there's a GCC build, would you expect it to defined `NON_MATCHING`? I'd rather it not have to.

Reply:
A GCC build will not be matching, so yeah. I could see replacing all NON_MATCHING with !MUST_MATCH, but having both is redundant.

Reply:
There's just a lot of code (in this PR but also going forward) where you just want `#if defined(MWERKS_GEKKO) && !defined(NON_MATCHING)`. So I don't really like `NON_MATCHING` anymore.

Replacing all of them seems a bit out of scope, but I can do it now if you prefer.

Reply:
To me !defined(NON_MATCHING) requires MWERKS_GEKKO. You could enforce this in a header with
```c
#if !defined(NON_MATCHING) && !defined(MWERKS_GEKKO)
#error "A matching build requires MWerks Gekko support"
#endif
```

Reply:
idk seems kind of hacky compared to just getting rid of it imo.

Reply:
Getting rid of what?

Reply:
Getting rid of `NON_MATCHING` and then there's just `MWERKS_GEKKO` (to access MWCC asm/features regardless of match) and `MUST_MATCH` for stuff that's only in the code because it needs to match.

I think `!MUST_MATCH` covers all the existing use cases of `NON_MATCHING`, which is basically for partial matches.

Reply:
The key difference being, that GCC (and Permuter, and clang, and M2C, etc etc) don't need to define `NON_MATCHING` or worry about an error. The symbols are all opt-in instead of declaring a negative.

Reply:
I think my main objection to seeing this was seeing two flags that are defined as opposites of each other, i.e. what happens when you compile with -DNON_MATCHING -DMUST_MATCH. There should be a single one that you provide (or don't provide) to the preprocessor, and the other is determined automatically via ifdefs.

After that, I suppose the choice of whether that flag is NON_MATCHING or MUST_MATCH is arbitrary. But every other decomp project uses NON_MATCHING, so I think we should stick to that.

Reply:
I see, well in general I don't really know (or care tbh) what other decomp projects are doing.

But anyway, in this case, if we agree that `NON_MATCHING` means "uses the original hardware and compiler but happens to not match" then we can continue to use that.